documentation/External-Grafana/ReadMe.md

```diff
@@ -0,0 +1,179 @@
+
+Currently Grafana operator has an integration to add Amazon Managed Service for Promethus (AMP) as a data source to Grafana instances hosted in a kubernetes environment. As more customers starting to use external grafana services, expanding the Grafana operator to support remote Grafana instances becomes inevitable. Adding ability to integrate with external grafana services, adding data sources, creating dashboards and alerting on a remote Grafana instances offloads responsibilities of managing a grafana instanaces from ops personas which helps them to focus on developing the features required for their business. This helps the customer teams to move from self managed Grafana instance on their Kubernetes environments to Pay as you go model on Grafana instances provided by providers.
+
+## Out of scope
+
+In the design, integration to remote managed Grafana providers other than Amazon Managed Grafana is out of scope at this point. We may add additional integrations as part of future road map items.
+
+## Verification
+
+- Create integration tests for adding keys for remote Grafana instance.
+- Create integration tests to create Grafana dashboards on a remote Grafana Instance.
+- Create integration tests to add cloud data sources such as Amazon Managed Service for Prometheus, Amazon CloudWatch to remote Grafana instance.
+- Create integration tests to create alerting on a remote Grafana Instance.
+
+## Current
+
+Currently the grafana operator supports the following for only self managed Grafana Instance :
+- Adding remote data sources such as Amazon Managed Service for Prometheus.
+- Creating Dashboards.
+- Setting up alerting.
+- And Many more.
+
+## Proposal
+
+In short the proposal in this document is about enhancing the Grafana Operator to support the integration to managed grafana services. We would need to enhance the current version of Grafana Operator to support the following :
+
+- Defining external Grafana instance as Grafana source to Grafana operator.
+- Creating Grafana dashboards on external Grafana instance.
+- Adding data sources to external Grafana instance.
+- Creating alerting on external Grafana instance.
+
+### Defining external Grafana instance as Grafana source to Grafana operator.
+
+Today Grafana operator supports self managed grafana instance as a Grafana source to Grafana operator. With this new feature, we would like to add a new CRD by name `GrafanaInstance` which basically creates an identity to an external Grafana instance. 
+
+> CRD design of how a `GrafanaInstance` will look like. In this sample design, `url` should be preloaded in a ConfigMap and `grafana_api_key` should be loaded to a Secret. Choice of using external secrets or loading secrets manually is end user responsibility :
+
+```.yaml
```

Comment:
Example using the existing grafana CRD to talk to an external grafana instance.

```.yaml
apiVersion: grafana.integreatly.org/v1beta1
kind: Grafana
metadata:
  annotations:
  labels:
    app: grafana
  name: grafana
  namespace: grafana-operator-system
spec:
  external:
    url: <external grafana url, type string>
    admin_username: <type SecretKeySelector>
    admin_password: <type SecretKeySelector>
```

If external is defined we won't create any new instance and instead try to talk to the external one.
We will probably have to change some CRD definitions in the code and make some features optional but that should be okay.

Reply:
Done.